lib/bolt/cli.rb

```diff
@@ -379,710 +341,601 @@ def validate(options)
+    #
+    # Process the command.
+    #
+    # @param app [Bolt::Application] The application.
+    # @param command [String] The command.
+    # @param action [String, NilClass] The action.
+    # @param options [Hash] The CLI options.
+    #
+    private def process_command(app, command, action, options)
+      case command
+      when 'apply'
+        results = outputter.spin do
+          app.apply(options[:object], options[:targets], **options.slice(:code, :noop))
+        end
+        rerun.update(results)
+        app.shutdown
+        outputter.print_apply_result(results)
+        results.ok? ? SUCCESS : FAILURE
 
-      handler = Signal.trap :INT do |signo|
-        @logger.info(
-          "Exiting after receiving SIG#{Signal.signame(signo)} signal.#{message ? ' ' + message : ''}"
-        )
-        exit!
-      end
+      when 'command'
+        outputter.print_head
+        results = outputter.spin do
+          app.command_run(options[:object], options[:targets], **options.slice(:env_vars))
+        end
+        rerun.update(results)
+        app.shutdown
+        outputter.print_summary(results, results.elapsed_time)
+        results.ok? ? SUCCESS : FAILURE
+
+      when 'file'
+        case action
+        when 'download'
+          outputter.print_head
+          results = outputter.spin do
+            app.file_download(options[:object], options[:leftovers].first, options[:targets])
+          end
+          rerun.update(results)
+          app.shutdown
+          outputter.print_summary(results, results.elapsed_time)
+          results.ok? ? SUCCESS : FAILURE
+        when 'upload'
+          outputter.print_head
+          results = outputter.spin do
+            app.file_upload(options[:object], options[:leftovers].first, options[:targets])
+          end
+          rerun.update(results)
+          app.shutdown
+          outputter.print_summary(results, results.elapsed_time)
+          results.ok? ? SUCCESS : FAILURE
+        end
 
-      # Initialize inventory and targets. Errors here are better to catch early.
-      # options[:target_args] will contain a string/array version of the targeting options this is passed to plans
-      # options[:targets] will contain a resolved set of Target objects
-      unless %w[guide module project secret].include?(options[:subcommand]) ||
-             %w[convert new show].include?(options[:action]) ||
-             options[:plan_hierarchy]
-        update_targets(options)
-      end
+      when 'group'
+        outputter.print_groups(**app.group_show)
+        SUCCESS
 
-      screen = "#{options[:subcommand]}_#{options[:action]}"
-      # submit a different screen for `bolt task show` and `bolt task show foo`
-      if options[:action] == 'show' && options[:object]
-        screen += '_object'
-      end
+      when 'guide'
+        app.guide(options[:object], outputter)
+        SUCCESS
+
+      when 'inventory'
+        targets = app.inventory_show(options[:targets])
```

Comment:
Ah, this is a tricky one: we need to validate that multiple targeting options aren't passed here, since the dreaded `if` statement on line 348 doesn't check when `inventory show` is run since it *can* have no targeting options, but still can't have *multiple* targeting options. I think that check can go here similar to how the other check is done in this class.

```
lucy ~/githubs/bolt ((afd2cdd0...)) $ be bolt inventory show --targets localhost --rerun all
Targets
  localhost    (Not found in inventory file)

Inventory source
  /home/lucy/githubs/bolt/inventory.yaml

Target count
  1 total, 0 from inventory, 1 adhoc

Additional information
  Use the '--detail' option to view target configuration and data
lucy ~/githubs/bolt ((afd2cdd0...)) $ bolt inventory show --targets localhost --rerun all
Only one targeting option '--targets', '--rerun', or '--query' can be specified
```

Reply:
I think this can be solved by just checking for multiple targeting options _before_ checking the specific command being run. Updated.